test/model/binding.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:flutter/foundation.dart';
+import 'package:zulip/model/binding.dart';
+import 'package:zulip/model/store.dart';
+
+import 'test_store.dart';
+
+/// A concrete binding for use in the `flutter test` environment.
+///
+/// Tests that will mount a [GlobalStoreWidget] should initialize this binding
+/// by calling [ensureInitialized] at the start of the `main` method.
+///
+/// Individual test functions that mount a [GlobalStoreWidget] may then use
+/// [globalStore] to access the global store provided to the [GlobalStoreWidget],
+/// and [TestGlobalStore.add] to set up test data there.  Such test functions
+/// must also call [reset] to clean up the global store.
+///
+/// The global store returned by [loadGlobalStore], and consequently by
+/// [GlobalStoreWidget.of] in application code, will be a [TestGlobalStore].
+class TestDataBinding extends DataBinding {
+  /// Initialize the binding if necessary, and ensure it is a [TestDataBinding].
+  ///
+  /// This method is idempotent; calling it repeatedly simply returns the
+  /// existing binding.
+  ///
+  /// If there is an existing binding but it is not a [TestDataBinding],
+  /// this method throws an error.
+  static TestDataBinding ensureInitialized() {
+    if (_instance == null) {
+      TestDataBinding();
+    }
+    return instance;
+  }
```

Comment:
How is this part accomplished:

```dart
  /// If there is an existing binding but it is not a [TestDataBinding],
  /// this method throws an error.
```

? In the case of `LiveDataBinding.ensureInitialized`, I see it's done with `as LiveDataBinding`, right:

```dart
  /// Initialize the binding if necessary, and ensure it is a [LiveDataBinding].
  static LiveDataBinding ensureInitialized() {
    if (DataBinding._instance == null) {
      LiveDataBinding();
    }
    return DataBinding.instance as LiveDataBinding;
  }
```

(Although, there, I see the error isn't mentioned in the method's doc; perhaps it should be? Ditto that the method is idempotent. …Actually, though, both points are probably less worth highlighting in the live-code method, since we only expect it to be called once. That'd be a fine reason to omit the points from the doc, I guess.)

Reply:
> How is this part accomplished:

In that case `TestDataBinding._instance` will be null (because there is no existing TestDataBinding, because the one existing binding is not a TestDataBinding), so the method body will construct `TestDataBinding()`. But then `DataBinding._instance` will be non-null (because there is some existing binding), so the assertion in the base class's constructor will fail.

Flutter's own BindingBase accomplishes a similar result in a slightly more elaborate way, with a `static Type? _debugInitializedType`:
```dart
  BindingBase() {
    // …
    assert(_debugInitializedType == null, 'Binding is already initialized to $_debugInitializedType');
    initInstances();
    assert(_debugInitializedType != null);
```

That seems potentially quite helpful in a world of lots of different potential binding classes — gives better error messages for sorting out bugs in that context — but since we really just have the one base class and two subclasses, and it's not clear we'll need more, I figured I'd keep it simple.